cookbooks/bcpc-hadoop/templates/default/hdp_core-site.xml.erb

```diff
@@ -120,4 +120,55 @@
     <name>hadoop.user.group.static.mapping.overrides</name>
     <value>hdfs=hadoop,hdfs;yarn=mapred,hadoop;</value>
   </property>
+  <% if node[:bcpc][:hadoop][:hdfs][:ldap][:integration] == true %>
+    <% if (node.run_list.expand(node.chef_environment).recipes.include?("bcpc-hadoop::namenode_master") ||
+      node.run_list.expand(node.chef_environment).recipes.include?("bcpc-hadoop::namenode_no_HA") ||
+      node.run_list.expand(node.chef_environment).recipes.include?("bcpc-hadoop::namenode_standby")) then %>
+   <!-- HDFS LDAP Mapping configuration -->
+  <property>
+    <name>hadoop.security.group.mapping</name>
+    <value>org.apache.hadoop.security.LdapGroupsMapping</value>
+  </property>
+
+  <property>
+    <name>hadoop.security.group.mapping.ldap.bind.user</name>
+    <value><%= node[:bcpc][:hadoop][:hdfs][:ldap][:user] %></value>
+  </property>
+
+  <property>
+    <name>hadoop.security.group.mapping.ldap.bind.password.file</name>
+    <value>/etc/hadoop/conf/ldap-conn-pass.txt</value>
+  </property>
+
+  <property>
+    <name>hadoop.security.group.mapping.ldap.url</name>
+    <value>ldap://<%= node[:bcpc][:hadoop][:hdfs][:ldap][:domain] %>:<%= node[:bcpc][:hadoop][:hdfs][:ldap][:port] %></value>
+  </property>
+
+  <property>
+    <name>hadoop.security.group.mapping.ldap.base</name>
+    <value>DC=<%= node[:bcpc][:hadoop][:hdfs][:ldap][:domain].split('.')[0] %>,DC=<%= node[:bcpc][:hadoop][:hdfs][:ldap][:domain].split('.')[1] %>,DC=<%= node[:bcpc][:hadoop][:hdfs][:ldap][:domain].split('.')[2] %></value>
+  </property>
+
+  <property>
+    <name>hadoop.security.group.mapping.ldap.search.filter.user</name>
+    <value>(&amp;(objectclass=user)(sAMAccountName={0}))</value>
```

Comment:
The search filters seem like they should be attributes; I can think of all sorts of uses that would vary depending on one's environment.